include/nodejs/node_osrm_support.hpp

```diff
@@ -138,6 +139,34 @@ inline engine_config_ptr argumentsToEngineConfig(const Nan::FunctionCallbackInfo
         return engine_config_ptr();
     }
 
+    auto algorithm = params->Get(Nan::New("algorithm").ToLocalChecked());
+    if (algorithm->IsString())
+    {
+        auto algorithm_str = Nan::To<v8::String>(algorithm).ToLocalChecked();
+        if (*v8::String::Utf8Value(algorithm_str) == std::string("CH"))
+        {
+            engine_config->algorithm = osrm::EngineConfig::Algorithm::CH;
+        }
+        else if (*v8::String::Utf8Value(algorithm_str) == std::string("CoreCH"))
+        {
+            engine_config->algorithm = osrm::EngineConfig::Algorithm::CoreCH;
+        }
+        else if (*v8::String::Utf8Value(algorithm_str) == std::string("MLD"))
+        {
+            engine_config->algorithm = osrm::EngineConfig::Algorithm::MLD;
+        }
+        else
+        {
+            Nan::ThrowError("algorithm option must be one of 'CH', 'CoreCH', or 'MLD'.");
+            return engine_config_ptr();
+        }
+    }
+    else if (!algorithm->IsUndefined())
+    {
+        Nan::ThrowError("algorithm option must be a string and one of 'CH', 'CoreCH', or 'MLD'.");
+        return engine_config_ptr();
+    }
```

Comment:
Hrm the docs say that the result from `Object::Get` is a `MaybeLocal<>` but that type does not have a `operator->`. For a `Local<>` calling `operator->` is not safe since it would dereference a `nullptr`. So yes this would need a check.